xds/internal/balancer/clusterresolver/balancerconfig/type.go

```diff
@@ -95,4 +95,50 @@ type DiscoveryMechanism struct {
 	// This is used for EDS watch if set. If unset, Cluster is used for EDS
 	// watch.
 	EDSServiceName string `json:"edsServiceName,omitempty"`
+	// DNSHostname is the DNS name to resolve in "host:port" form. For type
+	// LOGICAL_DNS only.
+	DNSHostname string `json:"dnsHostname,omitempty"`
+}
+
+// Equal returns whether the DiscoveryMechanism is the same with the parameter.
+func (dm DiscoveryMechanism) Equal(b DiscoveryMechanism) bool {
+	if dm.Cluster != b.Cluster {
+		return false
+	}
+	if !equalStringP(dm.LoadReportingServerName, b.LoadReportingServerName) {
+		return false
+	}
+	if !equalUint32P(dm.MaxConcurrentRequests, b.MaxConcurrentRequests) {
+		return false
+	}
+	if dm.Type != b.Type {
+		return false
+	}
+	if dm.EDSServiceName != b.EDSServiceName {
+		return false
+	}
+	if dm.DNSHostname != b.DNSHostname {
+		return false
+	}
```

Comment:
Using a `switch` here might be a little less verbose.

Reply:
Done